website_sale_charge_payment_fee/models/sale.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+
+from odoo import models, fields
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    def update_fee_line(self, acquirer):
+        for line in self.order_line:
+            if line.payment_fee_line:
+                line.unlink()
+        if acquirer.charge_fee:
+            if acquirer.charge_fee_type == 'fixed':
+                price = acquirer.charge_fee_fixed_price
+                if (
+                    self.company_id.currency_id.id !=
+                    self.pricelist_id.currency_id.id
+                ):
+                    price = self.company_id.currency_id.with_context(
+                        date=self.date_order
+                    ).compute(price, self.pricelist_id.currency_id)
+            elif acquirer.charge_fee_type == 'percentage':
+                amount_total_wo_delivery = sum(
+                    line.price_total for line in self.order_line if not line.is_delivery
```

Comment:
@AungKoKoLin1997 We can't do this without depending on the delivery module.

One way to avoid the dependency is to include something similar to https://github.com/qrtl/mi7-custom/tree/15.0/sale_order_payment_charge_line inside this module.

Reply:
I just added the dependency because expect this behavior to use with delivery module.